src/Dodgeball.ts

```diff
@@ -66,37 +68,119 @@ export class Dodgeball {
     return response;
   }
 
+  createErrorResponse(code: number, message: string){
+    return {
+      success: false,
+      errors: [{code: code, message: message}],
+      version: ApiVersion.v1,
+      verification: {
+        id: "",
+        status: VerificationStatus.FAILED,
+        outcome: VerificationOutcome.ERROR
+      }
+    }
+  }
+
   public async verify({
     workflow,
     dodgeballId,
     useVerification,
     options,
   }: IVerifyOptions): Promise<IDodgeballVerifyResponse> {
-    // Verify the event, expecting an acknowledgement containing details for tracking the asynchronous response
-    const response = (await makeRequest({
-      url: `${constructApiUrl(
-        this.config.apiUrl as string,
-        this.config.apiVersion
-      )}verify`,
-      method: "POST",
-      headers: constructApiHeaders(
-        this.secretKey,
-        useVerification?.id,
-        dodgeballId
-      ),
-      data: {
-        event: workflow,
-        options: options,
-      },
-    })) as IDodgeballVerifyResponse;
 
-    return response;
+    let trivialTimeout = !options.timeout || options.timeout <= 0;
+    let largeTimeout = options.timeout && options.timeout > 5*BASE_VERIFY_TIMEOUT_MS
+    let mustPoll = trivialTimeout || largeTimeout
+    let activeTimeout = mustPoll? BASE_VERIFY_TIMEOUT_MS: options.timeout ?? BASE_VERIFY_TIMEOUT_MS
```

Comment:
Can you add a space before the ```?``` and ```:``` here? Also, could you add parenthesis around the ```(options.timeout ?? BASE_VERIFY_TIMEOUT_MS)``` for clarity?